src/core/managers/workload.py

```diff
@@ -0,0 +1,107 @@
+from enum import Enum
+
+from abc import ABC, abstractmethod
+
+
+class WorkloadType(Enum):
+    POLKADOT_BINARY = 'polkadot_binary'
+    POLKADOT_SNAP = 'polkadot_snap'
```

Comment:
Nit: having "polkadot" in the naming here is arbitrary; for one everything in this charm already "is polkadot" and if every option has the same prefix it doesn't add anything of value.

```suggestion
class WorkloadType(Enum):
    BINARY = 'binary'
    SNAP = 'snap'
```

Reply:
done